be/dtm0_log.h

```diff
@@ -519,6 +535,44 @@ M0_INTERNAL void m0_be_dtm0_volatile_log_update(struct m0_be_dtm0_log  *log,
 M0_INTERNAL void m0_be_dtm0_log_pmsg_post(struct m0_be_dtm0_log *log,
 					  struct m0_fop         *fop);
 
+
+/**
+ * DTM0 Log iterator
+ *
+ * Provides a way to iterate over the DTM0 Log structure. Today has a very
+ * inefficient implementation and used to glue log related code together with
+ * redo logic and expose interfaces which hopefully will not change.
+ */
+struct m0_be_dtm0_log_iter {
+	struct m0_be_dtm0_log *dli_log;
+	struct m0_dtm0_tid     dli_current_tid;
+};
+
+M0_INTERNAL int m0_dtm0_log_rec_copy(struct m0_dtm0_log_rec       *dst,
+				     const struct m0_dtm0_log_rec *src);
+/**
+ * Used to finalise record returned from the iterator. Could be generalised in
+ * future.
+ */
+M0_INTERNAL void m0_dtm0_log_iter_rec_fini(struct m0_dtm0_log_rec *rec);
+
+M0_INTERNAL void m0_be_dtm0_log_iter_init(struct m0_be_dtm0_log_iter *iter,
+					  struct m0_be_dtm0_log      *log);
+
+M0_INTERNAL void m0_be_dtm0_log_iter_fini(struct m0_be_dtm0_log_iter *iter);
+
+/**
+ * Returns next record of the iterator pointing to the log.
+ *
+ * @param out returned record which is copied and need to be freed with
+ *            m0_dtm0_log_iter_rec_fini()
+ *
+ * @return +1 when the iterator was successfully moved to the next record.
+ * @return  0 when the end of the log has been reached.
+ */
+M0_INTERNAL int m0_be_dtm0_log_iter_next(struct m0_be_dtm0_log_iter *iter,
+					 struct m0_dtm0_log_rec	    *out);
```

Comment:
The return value here is not Linux conventional:
0 means success.
any other value means failure (including EOF)

Reply:
@nikitadanilov - I already made the change based on Hua's recommendation. Have a look at it.